### ISSUE TITLE
Custom Training gives no detections

### ISSUE THREAD

**Issue by icklerly1** [LSH9832/edgeyolo#15]
Hi, 
I am training edgeyolo on a custom dataset. Unfortunately the model does not detect any objects. 
Also the training loss does not decrease as expected.

number of images:
  train images: 15000
  val images: 4370

and the auto calculated anchors in yolov4 would be:
[5, 10],  [13, 30],  [25, 64],  [38,123],  [51,230],  [87,158],  [82,340], [136,420], [256,492]

My questions now:

Do I need to edit the anchors in _params/model/edgeyolo.yaml_ ?
And do I need to change the learning rate maybe?

**Comment by LSH9832** [comment]
please do not change anchors in yaml file because this is an anchor-free detector(which is equivalent to that the size of the anchor box is the reciprocal of the scale of the feature map) and changing to anchor-based models are not supported. you can change your learning rate if necessary.

**Comment by icklerly1** [comment]
Alright. 
So I want to adapt the learning rate:

# learning rate---------------------------------------------------------------------------------------------------------
lr_per_img: 0.00015625                               # total_lr = lr_per_img * batch_size_per_gpu * len(devices)

What is the desired total_lr ?

**Comment by LSH9832** [comment]
Generally 0.01 or less

**Comment by LSH9832** [comment]
there is no fixed way to adjust learning rate, it depends on yourself

**Comment by icklerly1** [comment]
Hi,

thank you. I now tried to adjusted the learning rate to a lower value.

I additionally checked our annotations as well. 
Below you can see a picture: 

![testAnno](https://user-images.githubusercontent.com/88317362/223159694-a3b8edae-4876-448a-96b0-9e9f98974a5b.jpg)

We only have 2 classes: "head" and "person".

In addition I changed the number of classes in params/model/edgeyolo.yaml:

from nc: 80 to nc: 2

I can start my training, the loss is decreasing a bit but right now I only get weird looking results, like this:

Do you have any idea what might be going wrong? I am really out of ideas..

![blub](https://user-images.githubusercontent.com/88317362/223161023-a317f3fc-afb9-4050-af77-f11c4762795d.jpg)

**Comment by LSH9832** [comment]
1. you can modify demo/load_dataset.py to ensure that images and annotations in your training is correct.
2. it is not necessary to change nc from 80 to 2, it finally depends on length of names in your dataset config.
3. it is recomended to use my model trained on coco dataset as your pretrained model

**Comment by icklerly1** [comment]
Thanks a lot for your help! 
I found an error in my dataset preparation.

It now works great!
Thanks again.
